impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Funnel.js

```diff
@@ -87,7 +87,8 @@ define([
         records.push({
           name: dataTable.getCompositeFormattedValue(i, categoriesColIndexes, this.groupedLabelSeparator),
           value: dataTable.getValue(i, measureColIndex),
-          tooltip: this._buildTooltip(this._buildRowTooltipHtml(dataTable, i), font)
+          tooltip: this._buildTooltip(this._buildRowTooltipHtml(dataTable, i), font),
+          visualKey: dataTable.getCompositeValue(i, categoriesColIndexes, this.groupedLabelSeparator)
```

Comment:
Should use the value separator, not the grouped label separator for this purpose, imo.

However, unless the echarts library poses any constraints on the type of the values of these "extension" properties, I would prefer the value would be directly an array, avoiding the issue of using the correct separator and serialization/deserialization altogether. You could use the `#getRowCells()` method. Each cell in the array has a `value` and `formatted` properties.

Reply:
Will try using `#getRowCells()`. Thankyou!!

Reply:
Hello @dcleao , I tried `#getRowCells()`. Works good for funnel and gauge, but for radar [visualKey](https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Radar.js#L176) is already of the format `key1~key2~..keyN`, which requires me to split using grouped label separator. 

So, should I proceed this way?

Reply:
Replied in the new PR: https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1766#discussion_r1603761485.